playback/src/decoder/passthrough_decoder.rs

```diff
@@ -32,19 +32,19 @@ where
     Ok(pck.data.into_boxed_slice())
 }
 
-pub struct PassthroughDecoder<R: Read + Seek> {
+pub struct PassthroughDecoder<'a, R: Read + Seek> {
     rdr: PacketReader<R>,
-    wtr: PacketWriter<Vec<u8>>,
+    wtr: PacketWriter<'a, Vec<u8>>,
     eos: bool,
     bos: bool,
     ofsgp_page: u64,
     stream_serial: u32,
-    ident: Box<[u8]>,
-    comment: Box<[u8]>,
-    setup: Box<[u8]>,
+    ident: u8,
+    comment: u8,
+    setup: u8,
```

Comment:
I'm not sure, why you're trying to change that into single `u8` values. You should be able to change that `Box<[u8]>` to `Vec<u8>` and the return type of `get_header` as well, and it should more or less compile. (Since `[T]::Owned` is `Vec<T>` and `.write_packet` is expecting a `Into<Cow<[u8]>>`.)

Reply:
Thanks this worked.